src/main/java/fr/utbm/info/vi51/worldswar/environment/Colony.java

```diff
@@ -0,0 +1,61 @@
+package fr.utbm.info.vi51.worldswar.environment;
+
+/**
+ * An ant colony is a group of allies ants. Ants from the same colony will
+ * cooperate for their colony's survival, and compete against any other colony.
+ *
+ */
+public class Colony {
+
+	/**
+	 * Unique identifier of the colony in the environment.
+	 */
+	private int id;
+
+	private Breed breed;
+
+	/**
+	 * Create a new Colony
+	 * 
+	 * @param id
+	 *            an integer that must be a unique identifier of the colony for
+	 *            the simulation
+	 * @param breed
+	 *            the {@link Breed} of the ants of the colony
+	 */
+	private Colony(int id, Breed breed) {
+		this.id = id;
+		this.breed = breed;
+	}
+
+	public Breed getBreed() {
+		return this.breed;
+	}
+
+	public void setBreed(Breed breed) {
+		this.breed = breed;
+	}
+
+	@Override
+	public int hashCode() {
```

Comment:
C'est une méthode hashCode générée par eclipse. C'est peut-être une erreur de l'avoir mise en fait... En gros je voulais qu'on puisse faire des `this.colony == otherAnt.colony` au lieu de `this.colony.getId() == otherAnd.colony.getId()`. Celà dit si l'on n'instancie chaque colonie qu'une fois on n'en a pas besoin et on pourrait même éventuellement se débarrasser d'id.
Si ça vous semble mieux je changerai ça